src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -920,6 +920,10 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
     void TermControl::_BlinkCursor(Windows::Foundation::IInspectable const& /* sender */,
                                    Windows::Foundation::IInspectable const& /* e */)
     {
+        if (!_terminal->IsCursorBlinkingAllowed() && _terminal->IsCursorVisible())

```

Comment:
Just a thought: when we change the `Terminal`'s cursor from blinking to not blinking (or vice-versa), should we trigger some sort of callback on `TermControl` to tell the cursor timer to stop (start)?

That might be over-engineering this though....

Reply:
Or have the new cursor design take affect on the next blink, so the timing is not interrupted?

Reply:
@zadjii-msft That's another way to stop the blinking. We'll just have to make sure that some other code flow doesn't start the timer again without checking whether blinking is allowed. Seems doable to me. Apart from the small perf benefit of not calling `_BlinkCursor` at every tick when blinking is off, I am not sure about the implications of this. So, I'll leave it up to you guys to decide.

Reply:
@mdtauk Can you please elaborate? How is the timing interrupted now?

Reply:
@dhowett-msft thoughts on this? I'm fine shipping this the way it is currently. It might be more confusing to have the `GotFocus` and `LostFocus` events check if blinking is allowed before manipulating the timer

Reply:
the cursor enablement state could be a common method called from `GotFocus`, `LostFocus` _and_  the thing that tells us the cursor type changed<sup>1</sup>.

however: we _do not have_ the event [1], so this is fine as is.